File: jobs/java_load/src/main/java/Load.java
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

import java.io.*;

public class Load implements Serializable {
	String filePath;

	SparkSession ss;

	Dataset<Row> items;

	public Load() {
		this.filePath = System.getenv("filePath");
		ss = SparkSession.builder().appName("Transform Job").master("local[*]").getOrCreate();
	}

	public void load() {

		items = ss.read().parquet(filePath + "transformed");

		items.write()
				.format("jdbc")
				.option("truncate", true)
				.option("driver", "org.postgresql.Driver")
				.option("url", System.getenv("url"))
				.option("dbtable", "vacancies")
				.option("user", System.getenv("user"))
				.option("password", System.getenv("password"))
				.mode("append")
				.save();

        ss.stop();
	}
}

File: jobs/scala_ETL_project/InfraStructure/core/src/main/scala/EL/Load.scala
package EL

import com.Config.CommonConfig
import com.Config.FolderName.FolderName
import org.apache.spark.sql.{DataFrame, SaveMode}

import scala.util.Try

object Load extends Serializable {
  def give(
            conf: CommonConfig,
            folderName: FolderName,
            data: DataFrame,
            format: String = "parquet"
          ): Try[Unit] = {
    Try(
      data.write
      .mode(SaveMode.Overwrite)
      .format(format)
      .save(conf.fs.getPath(folderName))
    )
  }
}